src/shared/modules/auth/common.js

```diff
@@ -0,0 +1,217 @@
+import jwtDecode from 'jwt-decode'
+import { isObject } from 'lodash'
+import {
+  AUTH_STORAGE_SSO_PROVIDERS,
+  AUTH_STORAGE_URL_SEARCH_PARAMS,
+  REDIRECT_URI
+} from './constants'
+import { addSearchParamsInBrowserHistory, authLog, authDebug } from './helpers'
+import {
+  defaultTokenTypeAuthentication,
+  defaultTokenTypePrincipal,
+  mandatoryKeysForSSOProviderParams,
+  mandatoryKeysForSSOProviders
+} from './settings'
+
+export const getInitialisationParameters = () => {
+  const urlSearchParams = window.location.search
+  const urlHashParamsAsSearchParams = '?' + window.location.hash.substring(1)
+
+  const initParams = {}
+  new URLSearchParams(urlSearchParams).forEach((value, key) => {
+    initParams[key] = value
+  })
+  new URLSearchParams(urlHashParamsAsSearchParams).forEach((value, key) => {
+    initParams[key] = value
+  })
+
+  return initParams
+}
+
+export const checkAndMergeSSOProviders = (
+  discoveredSSOProviders,
+  isLocalhostOrigin
+) => {
+  if (!discoveredSSOProviders || !discoveredSSOProviders.length) {
+    authLog('Invalid discovered SSO providers')
+    return
+  }
+
+  let currentSSOProviders =
+    JSON.parse(window.sessionStorage.getItem(AUTH_STORAGE_SSO_PROVIDERS)) || []
+  if (!Array.isArray(currentSSOProviders)) {
+    authLog(
+      'Found SSO providers in storage are defect, consider clearing the SSO provider state and reload',
+      'warn'
+    )
+    currentSSOProviders = []
+  }
+
+  discoveredSSOProviders.forEach(provider => {
+    if (!provider) return
+    if (
+      !mandatoryKeysForSSOProviders.every(key => provider.hasOwnProperty(key))
+    ) {
+      authLog(
+        `Dropping invalid discovered SSO provider with id: "${provider.id}", missing key`
+      )
+      return
+    }
+    if (
+      !mandatoryKeysForSSOProviderParams.every(key =>
+        provider?.params?.hasOwnProperty(key)
+      )
+    ) {
+      authLog(
+        `Dropping invalid discovered SSO provider with id: "${provider.id}", missing params key`
+      )
+      return
+    }
+    if (
+      currentSSOProviders.find(crntProvider => crntProvider.id === provider.id)
+    ) {
+      if (isLocalhostOrigin) {
```

Comment:
As per comment added to Bloom's PR, we should find out why this check is here, and why `isLocalhostOrigin` seems to always be set to true.